framework/log/Logger.php

```diff
@@ -115,13 +115,11 @@ class Logger extends Component
     public function init()
     {
         parent::init();
-        register_shutdown_function(function () {
-            // make regular flush before other shutdown functions, which allows session data collection and so on
-            $this->flush();
-            // make sure log entries written by shutdown functions are also flushed
-            // ensure "flush()" is called last when there are multiple shutdown functions
-            register_shutdown_function([$this, 'flush'], true);
-        });
+        // make regular flush before other shutdown functions, which allows session data collection and so on
```

Comment:
Perhaps it is actully better to have a public property for this instead of constants?
```php
// do something before session is closed
Yii::registerShutdownFunction($callback, [], Yii::$app->session->shutdownPriority - 1);
```
This will also allow manipulation on priorities via application configuration array.